src/algorithms/rhythm/harmonicbpm.cpp

```diff
@@ -39,14 +39,26 @@ void HarmonicBpm::configure() {
   _threshold = parameter("threshold").toReal();
   _bpm = parameter("bpm").toReal();
   _tolerance = parameter("tolerance").toReal();
+  // This patch was added to fix a crash that happens when a zero tolerance value is configured
+  if (_tolerance==0){ 
+    throw(EssentiaException("HarmonicBpm: Zero tolerance found in Configuration"));
+  }
+  if (_bpm==0){ 
+    throw(EssentiaException("HarmonicBpm: Zero bpm found in Configuration"));
+  }
```

Comment:
From the user point of view I disagree with having these exceptions.

1) The algorithm should return all BPMs harmonically related to a given bpm. BPM=0 can be simply ignored in the list of the input BPMs instead of having an exception. What is the specific issue of having it in the input?  We can log a `E_INFO` message with a warning of having encountered a value close to 0 BPM.

    Btw, our `bpm` parameter is only allowed to be >=1. Another option, for consistency, would be to have that type of check for what input values to ignore. Also because we don't want to mess with very small BPM values that hardly have any musical sense.

2) Tolerance 0 means an ideal harmonic match. This might be a valid parameter value we can expect to work. For example, for BPM=120, the input value 60 is an ideal match. The algorithm shouldn't hang in the first place. Can you pinpoint the specific place and reason why it happens? 

    I see that `areEqual` method will always return False on 0 tolerance: `std::fabs(error)<tolerance`: https://github.com/MTG/essentia/blob/master/src/essentia/utils/bpmutil.h#L72. This `fabs(...)  < 0` will never happen. We could compare instead with `fabs(...) <= max(tolerance, numeric_limits<Real>::epsilon())`.


By the way, I see that our `bpm` parameter is integer, but there is not specific reason for that. We can change it to `Real` for a more generic use-case.

Reply:
regarding the hanging of the program in when tolerance = 0, it occurs in  HarmonicBpm::findHarmonicBpms(...)
There are 2 while loops and the tolerance =0  results in the i++ never being called.
A "watchdog" solution could be added to avoid the while loop hanging, but that would only treat the syntoms,
Best solution is a new areEqual method.
If bpm (integer) is to be changed to  bpm (real), I prefer do that in separate commit. This would also require documentation update (https://essentia.upf.edu/reference/std_HarmonicBpm.html).

Reply:
Yes, sure
> If bpm (integer) is to be changed to bpm (real), I prefer do that in separate commit. This would also require documentation update (https://essentia.upf.edu/reference/std_HarmonicBpm.html).

In fact, I just checked again and this is only a cosmetic change for documentation, because the integer value is read into Real inside the configure method: `_bpm = parameter("bpm").toReal();`. Just change the default value `60` to `60.0` and the documentation should render the `Real` type for this parameter.